openmc/mgxs/mgxs.py

```diff
@@ -1884,13 +1886,19 @@ def tallies(self):
             energy = openmc.Filter('energy', group_edges)
             energyout = openmc.Filter('energyout', group_edges)
 
-            # Create a list of scores for each Tally to be created
-            if self.correction == 'P0':
-                scores = ['flux', 'scatter', 'scatter-P1']
-                filters = [[energy], [energy, energyout], [energyout]]
-            else:
-                scores = ['flux', 'scatter']
-                filters = [[energy], [energy, energyout]]
+            # Create lists of scores, filters for each Tally to be created
+            scores = ['flux']
+            filters = [[energy]]
+
+            # Create separate tallies for each moment
+            for moment in range(self.legendre_order+1):
+                scores.append('scatter-{}'.format(moment))
```

Comment:
Yes this was a dilemma for me and I'm certainly open to refactoring. The rationale behind this implementation is that it makes it simple to search for the tallies in the output. Right now if you specify a "scatter-PN" tally in your input, the statepoint/summary will expand the scores into a list of "scatter-0", "scatter-1", ... " scatter-N". Although this can be very nice for downstream data processing, it means that you cannot search for a tally in the output with the score "scatter-PN" that was used in the input. 

Of course I can refactor to get around this with a hack like the following:
* use "scatter-PN" in the input
* manually expand "scatter-PN" to "scatter-0", ... "scatter-N" when loading data from a statepoint
* slice apart the different moment tallies for ease of handling

Of course I should note that if one *merges*  the tallies in the input file (*i.e.*, `library.add_to_tallies_file(tallies_file, merge=True)`) then a single `Tally` will encapsulate all of the "scatter-0", ..., "scatter-N" scores even though they are separate in the `MGXS`. The tally merging was implemented to improve tallying efficiency. But perhaps a "scatter-PN" score is more efficient than a single tally with multiple "scatter-X" scores - perhaps you or @paulromano can clarify if this is indeed the case.